### PR TITLE
Azurecli batch

### PR DESCRIPTION
**Pull request by carlrab**
## Checklist

- [ ] Scripts in this pull request are written for the `bash` shell.
- [ ] This pull request was tested on __at least one of__ the following platforms:
  - [ ] Linux
  - [ ] Azure Cloud Shell
  - [ ] macOS
  - [ ] Windows Subsystem for Linux
- [ ] Scripts do not contain passwords or other secret tokens.
- [ ] No deprecated commands or arguments are used. ([Release notes](https://docs.microsoft.com/cli/azure/release-notes-azure-cli))
- [ ] All Azure resource identifiers which must be universally unique are guaranteed to be so.

### Testing information

CLI version:
```
az --version
```

Extensions required:

### CODE REVIEW

File: batch/add-application/add-application.sh [hunk 1]
```diff
@@ -1,49 +1,80 @@
...
 az batch application create \
-    --resource-group myResourceGroup \
-    --name mybatchaccount \
-    --application-name "My Application"
+    --resource-group $resourceGroup \
+    --name $batchAccount \
+    --application-name "MyApplication" # ?? must exist?
+
+# Error message
+(InvalidUri) The requested URI does not represent any resource on the server.
```
Comment (NickKouds) [inline batch/add-application/add-application.sh:44]: Application names cannot contain spaces, so removing the empty space should resolve the InvalidURI error.

File: batch/manage-pool/manage-pool-linux.sh [hunk 2]
```diff
@@ -1,32 +1,71 @@
...
-# optional and defaults to 'latest'.
+# optional and defaults to 'latest'."
+
 az batch pool create \
     --id mypool-linux \
     --vm-size Standard_A1 \
-    --image canonical:ubuntuserver:16.04-LTS \
-    --node-agent-sku-id "batch.node.ubuntu 16.04"
+    --image canonical:ubuntuserver:18.04-lts \
+    --node-agent-sku-id "batch.node.ubuntu 18.04" # what is query to get correct update to --image and --node-agent-ski-id?
```
Comment (NickKouds) [inline batch/manage-pool/manage-pool-linux.sh:65]: There is not a way to directly obtain the latest image/node-agent-sku-id. You can use the query parameter to at least filter on ubuntu support images i.e:
az batch pool supported-images list --query "[?contains(imageReference.offer,'ubuntuserver') && imageReference.publisher == 'canonical'].{Offer:imageReference.offer, Publisher:imageReference.publisher, Sku:imageReference.sku, nodeAgentSkuId:nodeAgentSkuId}[-1]" --output tsv

This should return the offer, publisher, and sku of the image along with the node agent sku id as a tsv and you can utilize further Bash processing to extract these values as variables and pass them to the AZ Batch CLI command.

File: batch/run-job/run-job.sh [hunk 3]
```diff
@@ -50,11 +61,14 @@ az batch task create \
     --job-id myjob \
     --json-file tasks.json
```
Comment (NickKouds) [inline batch/run-job/run-job.sh:63]: Regarding the error with the tasks.json file, you will need to pass the absolute path of the json file.

File: batch/manage-pool/manage-pool-windows.sh [hunk 4]
```diff
@@ -25,7 +48,13 @@ az batch pool create \
     --vm-size small \
     --start-task-command-line "cmd /c dir /s" \
     --start-task-wait-for-success \
-    --application-package-references myapp
+    --application-package-references myapp # ?? must exist?
```
Comment (NickKouds) [inline batch/manage-pool/manage-pool-windows.sh:51]: The value passed to the parameter **application-package-references** has to be the name of the Application followed by '#VersionNumber'. For example "MyApplication#1".

File: batch/create-account/create-account-user-subscription.sh [hunk 5]
```diff
@@ -1,40 +1,89 @@
...
+    --location "$location" \
     --enabled-for-deployment true \
     --enabled-for-disk-encryption true \
     --enabled-for-template-deployment true
 
 # Add an access policy to the Key Vault to allow access by the Batch Service.
 az keyvault set-policy \
-    --resource-group myResourceGroup \
-    --name mykevault \
-    --spn ddbf3205-c6bd-46ae-8127-60eb93363864 \
```
Comment (NickKouds) [inline batch/create-account/create-account-user-subscription.sh:25]: If you are creating a Batch account in User subscription mode, when you link the key vault, you have to specify the Microsoft Azure Batch (Application ID ddbf3205-c6bd-46ae-8127-60eb93363864) as the Service Principal. So specifying ddbf3205-c6bd-46ae-8127-60eb93363864 originally for the spn parameter should be what allows the Batch account to access the key vault.

**Comment by carlrab** [comment]
What is the absolute path for the JSON file in this repository?

**Pull request merged**